tt_metal/impl/device/device_pool.cpp

```diff
@@ -206,20 +206,31 @@ void DevicePool::initialize(
 
     // Never skip for TG Cluster
     bool skip = not tt::Cluster::instance().is_galaxy_cluster();
+    std::vector<chip_id_t> target_mmio_ids;
```

Comment:
We could reserve by device_ids.size() ahead of time? May slightly over-allocate but this vector is discarded after right?